include/zephyr/net/wifi_mgmt.h

```diff
@@ -402,6 +402,10 @@ struct wifi_connect_req_params {
 	uint8_t bssid[WIFI_MAC_ADDR_LEN];
 	/** Connect timeout in seconds, SYS_FOREVER_MS for no timeout */
 	int timeout;
+	/** PWE derivation */
+	int sae_pwe;
```

Comment:
As this is only for WPA3, IMHO we should extend the https://github.com/zephyrproject-rtos/zephyr/blob/4bd9522e1941073d8a942a36d526c27b158e91de/include/zephyr/net/wifi.h#L40 security type to include the variants

https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_supplicant.conf#n463 

* `WIFI_SECURITY_TYPE_SAE` = `WIFI_SECURITY_TYPE_SAE_HNP` 
* `WIFI_SECURITY_TYPE_SAE_H2E` 
* `WIFI_SECURITY_TYPE_SAE_AUTO` 

WDYT?

Reply:
No plan to add WIFI_SECURITY_TYPE_SAE_HNP, WIFI_SECURITY_TYPE_SAE_H2E, WIFI_SECURITY_TYPE_SAE_AUTO in wifi_security_type, we added WIFI_SECURITY_TYPE_SAE and sae_pwe type o support all above SAE types.

Reply:
But isn't it simpler to extend security types than adding another parameters specifically for SAE? We should separate API vs implementation, i.e., don't need to expose WPA supplicant internals 121 to the Wi-Fi management, we can abstract it out and then let `supp_api.c` translate to WPA supplicant calls.

Reply:
Updated as comments, pls help to review again, thanks!